tests/collections/asr/test_asr_metrics.py

```diff
@@ -73,56 +104,26 @@ def test_wer_function(self):
     def test_wer_metric_simple(self):
         wer = WER(vocabulary=self.vocabulary, batch_dim_index=0, use_cer=False, ctc_decode=True)
 
-        def get_wer(prediction: str, reference: str):
-            res = (
-                wer(
-                    predictions=self.__string_to_tensor(prediction),
-                    targets=self.__string_to_tensor(reference),
-                    target_lengths=torch.tensor([len(reference)]),
-                )
-                .detach()
-                .cpu()
-            )
-            return res[0] / res[1]
-
-        assert get_wer('cat', 'cot') == 1.0
-        assert get_wer('gpu', 'g p u') == 1.0
-        assert get_wer('g p u', 'gpu') == 3.0
-        assert get_wer('ducati motorcycle', 'motorcycle') == 1.0
-        assert get_wer('ducati motorcycle', 'ducuti motorcycle') == 0.5
-        assert get_wer('a f c', 'a b c') == 1.0 / 3.0
+        assert self.get_wer(wer, 'cat', 'cot') == 1.0
+        assert self.get_wer(wer, 'gpu', 'g p u') == 1.0
+        assert self.get_wer(wer, 'g p u', 'gpu') == 3.0
+        assert self.get_wer(wer, 'ducati motorcycle', 'motorcycle') == 1.0
+        assert self.get_wer(wer, 'ducati motorcycle', 'ducuti motorcycle') == 0.5
+        assert self.get_wer(wer, 'a f c', 'a b c') == 1.0 / 3.0
 
     @pytest.mark.unit
-    @pytest.mark.pleasefixme
     def test_wer_metric_randomized(self):
-        """This test relies on correctness of word_error_rate function"""
+        """This test relies on correctness of word_error_rate function."""
 
         def __randomString(N):
-            return ''.join(random.choice(string.ascii_lowercase + ' ') for i in range(N))
+            return ''.join(random.choice(''.join(self.vocabulary)) for i in range(N))
 
         wer = WER(vocabulary=self.vocabulary, batch_dim_index=0, use_cer=False, ctc_decode=True)
 
-        def get_wer(prediction: str, reference: str):
-            res = (
-                wer(
-                    predictions=self.__string_to_tensor(prediction),
-                    targets=self.__string_to_tensor(reference),
-                    target_lengths=torch.tensor([len(reference)]),
-                )
-                .detach()
-                .cpu()
-            )
-            return res[0] / res[1]
-
         for test_id in range(256):
             n1 = random.randint(0, 512)
```

Comment:
Please change to randint(1, 512) when both n1 and n2 are zero you get nan != Inf error

Reply:
fixed. Thanks!